.github/workflows/test.yml

```diff
@@ -0,0 +1,23 @@
+name: test
+on:
+  push:
+  pull_request:
+    types: [opened, synchronize]
+    branches: [main]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Checkout Repository"
+        uses: actions/checkout@v4
+      - name: Install node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 21.6.x
+          registry-url: 'https://registry.npmjs.org/'
+      - name: Install dependencies
+        run: npm ci
+      - name: Test
+        uses: coactions/setup-xvfb@v1
```

Comment:
Pls add a comment stating the reason for this

Reply:
done